pypesto/objective/roadrunner/roadrunner_calculator.py

```diff
@@ -0,0 +1,403 @@
+    ),
+    "log10_normal": lambda measurement, simulation, sigma: -0.5
+    * (
+        np.log(2 * np.pi * (sigma**2) * (measurement**2) * np.log(10) ** 2)
+        + ((np.log10(measurement) - np.log10(simulation)) / sigma) ** 2
+    ),
+    "lin_laplace": lambda measurement, simulation, sigma: -np.log(2 * sigma)
+    - (np.abs(measurement - simulation) / sigma),
+    "log_laplace": lambda measurement, simulation, sigma: -np.log(
+        2 * sigma * simulation
+    )
+    - (np.abs(np.log(measurement) - np.log(simulation)) / sigma),
+    "log10_laplace": lambda measurement, simulation, sigma: -np.log(
+        2 * sigma * simulation * np.log(10)
+    )
+    - (np.abs(np.log10(measurement) - np.log10(simulation)) / sigma),
+}
+
+
+class RoadRunnerCalculator:
+    """Class to handle RoadRunner simulation and obtain objective value."""
+
+    def __call__(
+        self,
+        x_dct: dict,  # TODO: sensi_order support
+        mode: ModeType,
+        roadrunner_instance: roadrunner.RoadRunner,
+        edatas: list[ExpData],
+        x_ids: Sequence[str],
+        parameter_mapping: list[ParMappingDictQuadruple],
+        petab_problem: petab.Problem,
+        solver_options: Optional[SolverOptions],
+    ):
+        """Perform the RoadRunner call and obtain objective function values.
+
+        Parameters
+        ----------
+        x_dct:
+            Parameter dictionary.
+        mode:
+            Mode of the call.
+        roadrunner_instance:
+            RoadRunner instance.
+        edatas:
+            List of ExpData.
+        x_ids:
+            Sequence of parameter IDs.
+        parameter_mapping:
+            Parameter parameter_mapping.
+        petab_problem:
+            PEtab problem.
+        solver_options:
+            Solver options of the roadrunner instance Integrator. These will
+            modify the roadrunner instance inplace.
+
+        Returns
+        -------
+        Tuple of objective function values.
+        """
+        # sanity check that edatas and conditions are consistent
+        if len(edatas) != len(parameter_mapping):
+            raise ValueError(
+                "Number of edatas and conditions are not consistent."
+            )
+        if solver_options is None:
+            solver_options = SolverOptions()
+        # apply solver options
+        solver_options.apply_to_roadrunner(roadrunner_instance)
+        simulation_results = {}
+        llh_tot = 0
+        for edata, mapping_per_condition in zip(edatas, parameter_mapping):
+            sim_res, llh = self.simulate_per_condition(
+                x_dct, roadrunner_instance, edata, mapping_per_condition
+            )
+            simulation_results[edata.condition_id] = sim_res
+            llh_tot += llh
+
+        if mode == MODE_FUN:
+            return {
+                "fval": -llh_tot,
+                "simulation_results": simulation_results,
+                "llh": llh_tot,
```

Comment:
Use pypesto constant FVAL? 
And make constants for the other two, if they don't exist already?